test/lib/cql_test_env.cc

```diff
@@ -771,13 +771,18 @@ class single_node_cql_env : public cql_test_env {
                                                      std::chrono::duration_cast<std::chrono::milliseconds>(cql3::prepared_statements_cache::entry_expiry));
             auth_prep_cache_config.refresh = std::chrono::milliseconds(cfg->permissions_update_interval_in_ms());
 
-            auto udf_enabled = cfg->enable_user_defined_functions() && cfg->check_experimental(db::experimental_features_t::feature::UDF);
-            std::shared_ptr<rust::Box<wasmtime::Engine>> wasm_engine;
-            if (udf_enabled) {
-                wasm_engine = std::make_shared<rust::Box<wasmtime::Engine>>(wasmtime::create_engine(cfg->wasm_udf_memory_limit()));
+            std::optional<wasm::startup_context> wasm_ctx;
+            if (cfg->enable_user_defined_functions() && cfg->check_experimental(db::experimental_features_t::feature::UDF)) {
+                wasm_ctx.emplace();
+                wasm_ctx->engine = std::make_shared<rust::Box<wasmtime::Engine>>(wasmtime::create_engine(cfg->wasm_udf_memory_limit()));
+                wasm_ctx->alien_runner = std::make_shared<wasm::alien_thread_runner>();
+                size_t max_cache_size = dbcfg.available_memory * cfg->wasm_cache_memory_fraction();
+                wasm_ctx->cache_size = max_cache_size;
+                wasm_ctx->instance_size = cfg->wasm_cache_instance_size_limit();
+                wasm_ctx->timer_period = std::chrono::milliseconds(cfg->wasm_cache_timeout_in_ms());
```

Comment:
I'd appreciate (as a followup) is struct wasm_ctx got a constructor accepting db::config (and likely database::config, it looks like it needs one) and initialize all this stuff on its own. This would generalize this code with main's similar one